pkg/rds/queries.go

```diff
@@ -283,25 +283,45 @@ func ApplyConfig(
 		Parameters:           modifiedParameters,
 	}
 
-	// TODO(eddie): We should actuall verify in the response that it worked
+	// TODO(eddie): We should actually verify in the response that it worked
 	_, err = clients.RDSClient.ModifyDBParameterGroup(ctx, args)
 	if err != nil {
-		return fmt.Errorf("failed to modify parameter group: %v", err)
+		return fmt.Errorf("failed to modify parameter group: %w", err)
 	}
 
 	// Wait for parameter group changes to be processed
 	logger.Info("Waiting for parameter group changes to be processed...")
-	err = waitRDSInstanceAvailable(clients, databaseIdentifier, parameterGroupName, ctx)
+	result, err := GetRDSInstanceWithParameterGroupStatus(clients, databaseIdentifier, parameterGroupName, ctx)
 	if err != nil {
-		return fmt.Errorf("error waiting for parameter group changes to be processed: %v", err)
```

Comment:
Old code just assumed once it was applied, it succeeded, we now explicitly check.